### PR TITLE
feat: user mailing list

### PR DESCRIPTION
**Pull request by sshanzel**
Migrated the workers from the gateway.

### CODE REVIEW

**Review by rebelchris — APPROVED** [review]
Nice work on this 🙏

File: src/common/mailing.ts [hunk 1]
```diff
@@ -91,3 +92,95 @@ export const getCommentedAuthorMailParams = (
     },
   };
 };
+
+// taken from sendgrid library itself
+type HttpMethod =
```
Comment (sshanzel) [inline src/common/mailing.ts:97]: The type-safety of sendgrid shouts when sending the parameters without casting the method to this type.

File: .infra/workers.ts [hunk 2]
```diff
@@ -19,6 +19,14 @@ export const workers: Worker[] = [
...
     subscription: 'user-updated-api',
   },
+  {
+    topic: 'user-updated',
+    subscription: 'user-updated-mailing',
+  },
+  {
+    topic: 'user-updated',
+    subscription: 'user-deleted-mailing',
+  },
```
Comment (idoshamun) [inline .infra/workers.ts:29]: I think these subs already exist in the gateway. Am I right?
If so, Pulumi will not be able to create them. We might need to do some manual work here. Let's think about it later

**Review by idoshamun — CHANGES_REQUESTED** [review]
I think we have to approach this PR a bit differently.
If we still rely on the events we receive from the gateway, we still rely on the gateway itself.
So I suggest publishing new events based on changes in the users table of the API and then update the mailing list accordingly.
What do you think? cc @rebelchris

**Comment by sshanzel** [comment]
> I think we have to approach this PR a bit differently.
If we still rely on the events we receive from the gateway, we still rely on the gateway itself.
So I suggest publishing new events based on changes in the user's table of the API and then update the mailing list accordingly.

I actually initially thought of publishing an event here in the API (but was missing in this PR), but now that we have kept the user change event in the gateway, and kept the changes minimal, we should not need an additional worker for the moment. WDYT?

**Comment by idoshamun** [comment]
Let's wait for the planning meeting and we decide :)

File: .infra/workers.ts [hunk 3]
```diff
@@ -19,6 +19,14 @@ export const workers: Worker[] = [
     topic: 'user-updated',
     subscription: 'user-updated-api',
   },
+  {
+    topic: 'user-updated',
+    subscription: 'user-updated-mailing',
+  },
+  {
+    topic: 'user-updated',
```
Comment (rebelchris) [inline .infra/workers.ts:27]: This should be user-deleted topic

**Review by rebelchris — APPROVED** [review]

**Pull request merged**